### DISCUSSION TITLE
Dev-Triggered action to automatically order and space achievements.

### DISCUSSION THREAD

**Discussion by redwizard42** [RetroAchievements/RAWeb#2117] [Ideas]
I would sometimes like the ability to automatically generate achievement order values.  Starting with the current sort order (which, if all 0 is the achievement ID as far as I can tell), I'd like to be able to enter a spacing and have all the achievements assigned based on their current order.

Example: I trigger the function, enter a spacing of 10, and confirm. Then the first achievement get assigned an order value of 10, then the next is 20, then 30, etc. Or some similar mechanism.

Why?
I almost always publish my achievements in order, but some emulators will display them in whatever order unless order values are assigned, so I end up having to assign them anyway. This would make this process easier and leave spaces in case a revision needs to slot achievements into the middle.
-OR-
I already have my set ordered, but a revision comes along that will need to slot into the middle anyway and there isn't already reserved spacing.  This would let me re-space things to prepare for the revision achievements.

**Comment by ladynadiad** [comment]
How would you want it to be handled though in cases like set is already ordered and it needs to be reordered or an achievement is being inserted?  I tend to upload a few achievements at a time and handle all marking and ordering as I go.  Maybe an option to be able to set it to a set number higher than the highest currently present would be a good addition (and save me having to scroll to the bottom to remember what my last number was).

Also what about stuff where you may have large sections of things like for example I had Shadow Hearts: Covenant going in blocks starting at particular parts to group achievements of certain types together.  Like all of the fusions were in one area, all the ring souls in another and so on.  By the end of that set I was in the thousands with all the various blocks of optional content that set has, but it was easy to insert a new achievement into an existing block because I left room between them but if it was set to automatically number stuff it could completely mess up the ordering and make it tricky to add to a specific middle block.  Also a similar thing was done with Smash Pack where each dev had a block of 1000 to work with for their stuff.  Obviously if I realized my achievements there were needing a fix to the order I wouldn't want to risk messing everyone else's up, but I'm pretty far down on the list in the 5000s.

I think this would be especially nice if there was a way to select a set of achievements, maybe checkboxes for it, and choose a starting number for it so we could also be able to do blocks and insert stuff into a specific block.  That's great when you're dealing with groups of similar achievements like a set of achievements for story mode and one for arcade mode in a fighting game where you might want to give story from 1-100 and then arcade 101-200 to allow for easy insertion later of a new story achievement while also still giving plenty of room for new ideas for story by putting arcade into a much later series of numbers.
